backend/routers/routes/posts.py

```diff
@@ -0,0 +1,73 @@
+from typing import Annotated, Literal
+
+from crud import post_crud, user_crud
+from fastapi import APIRouter, Depends, HTTPException, Query, Request, Response
+from pydantic import BaseModel, Field
+
+post_router = APIRouter()
+
+
+class FilterParams(BaseModel):
+    creator_email: str | None = None
+    sort: str | None = None
+
+
+class Post(BaseModel):
+    title: str
+    content: str
+    parent_id: str | None = None
+    tags: list[str]
+
+
+class PostEdit(BaseModel):
+    new_content: str
+    edit_reason: str | None = None
+
```

Comment:
thoughts on moving this to the `models` folder - (i noticed we were adding a bunch of these classes)
https://github.com/ubclaunchpad/forum/tree/dev/backend/models